src/main/java/kafka/vertx/demo/Main.java

```diff
@@ -35,8 +48,56 @@ public static void main(String[] args) {
     Future<String> periodicProducerDeployment = vertx.deployVerticle(new PeriodicProducer());
     Future<String> webSocketServerDeployment = vertx.deployVerticle(new WebSocketServer());
 
-    CompositeFuture.all(periodicProducerDeployment, webSocketServerDeployment)
+    CompositeFuture.join(periodicProducerDeployment, webSocketServerDeployment)
       .onSuccess(ok -> logger.info("✅ Application started in {}ms", currentTimeMillis() - startTime))
       .onFailure(err -> logger.error("❌ Application failed to start", err));
   }
+
+  public static Future<JsonObject> loadKafkaConfig(Vertx vertx, String propertiesPath) {
+    File propertiesPathFile = new File(propertiesPath);
+    String properties = propertiesPathFile.isDirectory() ? new File(propertiesPathFile, DEFAULT_PROPERTIES_PATH).toString() : propertiesPath;
+    ConfigRetriever configRetriever =  ConfigRetriever.create(vertx,
+      new ConfigRetrieverOptions().addStore(
+        new ConfigStoreOptions()
+          .setType("file")
+          .setFormat("properties")
+          .setConfig(new JsonObject().put("path", properties).put("raw-data", true)))).setConfigurationProcessor(configurationProcessor(properties));
+    FileSystem fileSystem = vertx.fileSystem();
+    return fileSystem.exists(properties)
+      .compose(exists -> {
+        if (exists) {
+          return configRetriever.getConfig();
+        } else {
+          return Future.failedFuture(String.format("Kafka properties file is missing. Either specify using -D%s=<path> or use the default path of %s.", PROPERTIES_PATH_ENV_NAME, DEFAULT_PROPERTIES_PATH));
```

Comment:
Log out the failed path too

Reply:
Also, I'm guessing docs for 4.0 are wrong as it says getConfig() returns void...

Reply:
Does it also throw errors if config cannot be parsed?

Reply:
It certainly seems to return the config, where in the docs does it say it returns void? I think it would be in the getConfig that it would fail if it couldn't parse it, so that would get caught higher up, but I haven't tried it.

Reply:
https://vertx-ci.github.io/vertx-4-preview/docs/apidocs/io/vertx/reactivex/config/ConfigRetriever.html#getConfig--

Reply:
Could we add the failed path to the failure message? `Kafka properties file is missing as location : x `